### ISSUE TITLE
Maizzle event beforeRender don't run when build is triggered after changes with maizzle serve

### ISSUE THREAD

**Issue by schirliuradu** [maizzle/framework#335]
- Maizzle Version: 2.3.0
- Node.js Version: 14.2.0

Hi Cosmin,

I wanted to let you know that i think i found a small problem with maizzle events that coukd be also related to hot reload. 

The problem is that if you make some changes in an event like it could be the ``beforeRender`` related to the html of every template that you parse (for example you make a regex on non compiled templates because you want to change something, but there are lots of use cases), the function / regex will be applied only on first time when you compile. 

I try to explain the fact in a bit better way: if you run maizzle build production, or just maizzle build for local building you don't see the issue, but, if you start dev server with maizzle serve than the first time everything will work and the function / regex added in beforeRender event will be considered and applied. However, when you go into a template file and make some changes (even empty space just for retriggering build process), that function / regex will not be considered more. I have to say that i noticed this problem ONLY when changing templates, and with this i mean that if i go into a component / layout file and make some changes and save the build process is retriggered and everything works fine. It happens only with the templates. 

I won't attach here an example because i think it's easy to reproduce this issue. Let me know if you have any idea about what could be the fact that causes this issue. 

Thank you.

**Comment by cossssmin** [comment]
Yeah, this is a side effect of a feature in 2.3.0 that only compiles the template you've saved, instead of compiling everything. This was added so that you get much faster build times when developing locally - things were getting a bit too slow in projects with tens of templates.

Maizzle still compiles everything (with the `build()` method) whenever you edit a watched file that is not a template, so that's why it still works for others like layouts or components.

I'm not sure how this can be fixed, but I wouldn't trade the new build speed improvement for something that can still be achieved by saving a non-template, watched file.

**Comment by cossssmin** [comment]
Actually, after a good night's sleep, the issue is that we're not passing events to `render` when using `maizzle serve`:

https://github.com/maizzle/framework/blob/0859a69fff02d1f70e28d132e25880af41df7377/src/index.js#L80-L85

That should actually be:

```js
await self.render(await fs.readFile(file, 'utf8'), {
  maizzle: config,
  tailwind: {
    css: cssString
  },
  ...config.events
})
```

Just did a test with this:

```js
events: {
  beforeRender(html, config) {
    config.timestamp = new Date()
    return html
  },
}
```

And now every time I save the template where I use `{{ page.timestamp }}`, the new date is rendered in the browser.

Will fix this is the next release, thanks for spotting it 👍

**Comment by cossssmin** [comment]
@schirliuradu published [v2.3.1](https://github.com/maizzle/framework/releases/tag/v2.3.1)
